tests/text/test_wer.py

```diff
@@ -14,55 +19,57 @@
 PREDICTION2 = "what a day"
 REFERENCE2 = "what a wonderful day"
 
-BATCH_PREDICTIONS = [PREDICTION1, PREDICTION2]
-BATCH_REFERENCES = [REFERENCE1, REFERENCE2]
-
-
-@pytest.mark.parametrize(
-    "prediction,reference",
-    [(PREDICTION1, REFERENCE1), (PREDICTION2, REFERENCE2)],
-)
-def test_wer_functional_single_sentence(prediction, reference):
-    """Test functional with strings as inputs."""
-    pl_output = wer(prediction, reference)
-    jiwer_output = compute_measures(reference, prediction)["wer"]
-    assert pl_output == jiwer_output
-
-
-def test_wer_functional_batch():
-    """Test functional with a batch of sentences."""
-    pl_output = wer(BATCH_PREDICTIONS, BATCH_REFERENCES)
-    jiwer_output = compute_measures(BATCH_REFERENCES, BATCH_PREDICTIONS)["wer"]
-    assert pl_output == jiwer_output
-
-
-@pytest.mark.parametrize(
-    "prediction,reference",
-    [(PREDICTION1, REFERENCE1), (PREDICTION2, REFERENCE2)],
-)
-def test_wer_class_single_sentence(prediction, reference):
-    """Test class with strings as inputs."""
-    metric = WER()
-    metric.update(prediction, reference)
-    pl_output = metric.compute()
-    jiwer_output = compute_measures(reference, prediction)["wer"]
-    assert pl_output == jiwer_output
-
-
-def test_wer_class_batch():
-    """Test class with a batch of sentences."""
-    metric = WER()
-    metric.update(BATCH_PREDICTIONS, BATCH_REFERENCES)
-    pl_output = metric.compute()
-    jiwer_output = compute_measures(BATCH_REFERENCES, BATCH_PREDICTIONS)["wer"]
-    assert pl_output == jiwer_output
-
-
-def test_wer_class_batches():
-    """Test class with two batches of sentences."""
-    metric = WER()
-    for prediction, reference in zip(BATCH_PREDICTIONS, BATCH_REFERENCES):
-        metric.update(prediction, reference)
-    pl_output = metric.compute()
-    jiwer_output = compute_measures(BATCH_REFERENCES, BATCH_PREDICTIONS)["wer"]
-    assert pl_output == jiwer_output
+BATCHES = {"preds": [[PREDICTION1], [PREDICTION2]], "targets": [[REFERENCE1], [REFERENCE2]]}
+
+
+def _compute_wer_metric_jiwer(prediction: Union[str, List[str]], reference: Union[str, List[str]]):
+    return compute_measures(reference, prediction)["wer"]
+
+
+@pytest.mark.skipif(not _JIWER_AVAILABLE, reason="test requires jiwer")
+class TestWER(TextTester):
```

Comment:
insert it here to parametrize all tests with preds and target one time
```suggestion
@pytest.mark.parametrize(
        ["preds", "targets"],
        [
            pytest.param(BATCHES["preds"], BATCHES["targets"]),
        ],
    )
class TestWER(TextTester):
```